agent/util-scripts/pbench-tool-meister-start

```diff
@@ -502,6 +507,9 @@ class ToolDataSink(BaseServer):
                 kind = data["kind"]
                 action = data["action"]
                 status = data["status"]
+                if status != "success":
+                    tms = json.loads(redis_client.get(tm_data_key))
+                    failure_count = log_failed_tool_init(logger, tms)
```

Comment:
Then again ... thinking more about Webb's comment on `kind` and `action` ... the "failed tool init" really only applies to the TDS confirmation of the init message, whereas this loop is processing a sequence of TDS messages, potentially including others?

I'm not really sure what we should be doing here. Are there any other messages aside from ds/startup we really expect to handle here? What should we do about them? I don't think we should be "logging tool failure" for any other sort of message, if there are any. Do we log them and discard them? Should we be prepared to take some additional action?

At the very least we should be checking what we're getting.

I really think the original expectation here was that we'd get a *single* message confirming startup through TDS, and maybe the initial `status = ""` was simply intended to handle the case where the generator doesn't yield any results. (And there's no exit from this loop; it exits when the Redis pubsub listener quits or throws.

In fact, I realize that I initially completely misunderstood the context and purpose of this loop, and now I'm not sure I understand how it works at all. What makes the pubsub channel "stop" here so that we terminate the generator and this loop? Clearly that happens somehow, and the structure now isn't significantly different than before.

So, yeah, the old `status` returned would be the status for the final *message* received through the pubsub channel before the generator stopped; and **if** that was the startup confirmation (which is clearly implicitly assumed by the loop), exiting with that status is fine: any previous messages are presumably irrelevant. (What would they be? Does this actually happen?)

Reply:
> Then again ... thinking more about Webb's comment on `kind` and `action` ... the "failed tool init" really only applies to the TDS confirmation of the init message, whereas this loop is processing a sequence of TDS messages, potentially including others?
> 
> I'm not really sure what we should be doing here. Are there any other messages aside from ds/startup we really expect to handle here? What should we do about them? I don't think we should be "logging tool failure" for any other sort of message, if there are any. Do we log them and discard them? Should we be prepared to take some additional action?
> 
> At the very least we should be checking what we're getting.
> 
> I really think the original expectation here was that we'd get a _single_ message confirming startup through TDS, and maybe the initial `status = ""` was simply intended to handle the case where the generator doesn't yield any results. (And there's no exit from this loop; it exits when the Redis pubsub listener quits or throws.
> 
> In fact, I realize that I initially completely misunderstood the context and purpose of this loop, and now I'm not sure I understand how it works at all. What makes the pubsub channel "stop" here so that we terminate the generator and this loop? Clearly that happens somehow, and the structure now isn't significantly different than before.
> 
> So, yeah, the old `status` returned would be the status for the final _message_ received through the pubsub channel before the generator stopped; and **if** that was the startup confirmation (which is clearly implicitly assumed by the loop), exiting with that status is fine: any previous messages are presumably irrelevant. (What would they be? Does this actually happen?)

I did a benchmark run, where all the requirements were met and it was successful without emitting anything other than these messages and also on failures. So, I am also not sure if there might be some other conditions that might provoke it to emit a message which I might not have seen.